lib/modules/login/login_presenter.dart

```diff
@@ -0,0 +1,25 @@
+part of 'login_module.dart';
+
+abstract class LoginPresenter
+    extends Presenter<LoginView, LoginInteractor, LoginRouter> {}
+
+class LoginPresenterImpl extends LoginPresenter
```

Comment:
I feel like this `LoginPresenterImpl extends LoginPresenter` is a bit odd. Wondering if you abstract it to provide later from your DI: 
`locator.registerFactory<LoginPresenter>(() => LoginPresenterImpl());` but its interface like `LoginPresenter ` doesn't define any behaviors different from the implementation class `LoginPresenterImpl`; then the abstraction here is kinda redundant imo? 🤔  Is there anything obstruct you if we just have:
```
class LoginPresenter extends Presenter<LoginView, LoginInteractor, LoginRouter>
    implements LoginViewDelegate, LoginInteractorDelegate {

}
``` 
? 🤔

Reply:
@sleepylee if we do follow this
```dart
class LoginPresenter extends Presenter<LoginView, LoginInteractor, LoginRouter>
    implements LoginViewDelegate, LoginInteractorDelegate {

}
```
I intent to hide all implements from `LoginViewDelegate`, `LoginInteractorDelegate`, so it make a call from presenter much clearer. 😁

Reply:
not sure if I understand it correctly 🤔  but later you're calling directly to the `LoginViewDelegate ` anyway, not anything directly from the `LoginPresenter` (as you're not having anything yet too). I'm still not sure if removing the extra abstract class declaration above could lead to anything different from the caller's side, or could you please enlighten me by some examples? 🤔

Reply:
@sleepylee I mean
If we do it
```dart
abstract LoginViewDelegate {
 BehaviorSubject get stateDidInit;
}

class LoginPresenter extends Presenter<LoginView, LoginInteractor, LoginRouter>
    implements LoginViewDelegate, LoginInteractorDelegate {

}
```
then `presenter. stateDidInit` show up in code suggestion 😁 
and our generated mock doesn't generate junk things

But, in fact, I think this is good to have but not required, IMO 😀

Reply:
I get this point:
>then presenter. stateDidInit show up in code suggestion 😁

as when you're providing the `Presenter` from DI, you returns the implemented class. But I'm still curious (as this is to make sure, later on, our convention/template we won't over-abstracting 😓 ), we're not allowing any `presenter` instance to be accessed from anywhere anw, so is it redundant to worry about this concern at all? 🤔  
From your architecture, I'm seeing that the presenter is only being used to configure at the base `Module`, there is no access provided for the implemented class 🤔  the presenter is managing the connection between the `Interactor` (business state provider) and the `Delegate` (the real caller when an event is happening). So `presenter` is actually just bridging without being instantiated anywhere 🤔  that's where I think that you may not need to add another abstraction on this, instead, when testing, you just need to instantiate it directly for testing purpose 🤔  it will ensure your coverage even higher 🤔 

or am I missing anything? 💭

Reply:
Alright, let me update it on another PR, cause this will change a lot of my current PRs 👍

Reply:
don't worry about it for now, @markgravity. If you agree with that then you can do the refactor work after finishing the main goal of the app 👍